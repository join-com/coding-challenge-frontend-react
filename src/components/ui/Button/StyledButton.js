import styled, { css } from "styled-components";
import { colors } from "../../../constants/styles";
import { defaultInputStyles } from "../../../helpers/stylesMixins";

const buttonHoverStyles = css`
  color: ${colors.primary};
`;

const StyledButton = styled.button`
  ${defaultInputStyles};
  text-align: center;
  margin-right: 0.5rem;
  &:hover {
    ${({ disabled }) => !disabled && buttonHoverStyles};
  }
`;

export default StyledButton;
