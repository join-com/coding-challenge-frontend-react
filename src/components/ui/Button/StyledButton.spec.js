import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import StyledButton from "./StyledButton";

describe("Styled StyledButton component", () => {
  const defaultProps = {
    uppercase: false,
    size: "normal",
    disabled: false
  };

  const getWrapper = (testProps = {}) => {
    const props = { ...defaultProps, ...testProps };
    return renderer.create(<StyledButton {...props} />).toJSON();
  };

  it("should match snapshot", () => {
    const wrapper = getWrapper();
    expect(wrapper).toMatchSnapshot();
  });

  it("should match snapshot with disabled prop", () => {
    const wrapper = getWrapper({ disabled: true });
    expect(wrapper).toMatchSnapshot();
  });
});
