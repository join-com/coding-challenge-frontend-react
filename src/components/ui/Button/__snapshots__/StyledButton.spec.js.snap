// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styled StyledButton component should match snapshot 1`] = `
.c0 {
  border: solid 2px #1D232B;
  border-radius: 3px;
  font-family: 'Oswald',sans-serif;
  font-size: 1rem;
  font-weight: bold;
  line-height: 1.2;
  background-color: #F9F9F9;
  color: #1D232B;
  opacity: 1;
  padding: 0.5rem;
  outline: none;
  -webkit-transition: all 0.1s ease-in;
  transition: all 0.1s ease-in;
  border-radius: 3px;
  text-transform: none;
  cursor: pointer;
  text-align: center;
  margin-right: 0.5rem;
}

.c0:hover {
  border-color: #90D3DC;
}

.c0:hover {
  color: #90D3DC;
}

<button
  className="c0"
  disabled={false}
  size="normal"
/>
`;

exports[`Styled StyledButton component should match snapshot with disabled prop 1`] = `
.c0 {
  border: solid 2px #1D232B;
  border-radius: 3px;
  font-family: 'Oswald',sans-serif;
  font-size: 1rem;
  font-weight: bold;
  line-height: 1.2;
  background-color: #F9F9F9;
  color: #1D232B;
  opacity: 0.5;
  padding: 0.5rem;
  outline: none;
  -webkit-transition: all 0.1s ease-in;
  transition: all 0.1s ease-in;
  border-radius: 3px;
  text-transform: none;
  cursor: not-allowed;
  text-align: center;
  margin-right: 0.5rem;
}

<button
  className="c0"
  disabled={true}
  size="normal"
/>
`;
